internal/worker.go

```diff
@@ -291,6 +297,25 @@ type (
 		// Deprecated: All bugports are always deprecated and may be removed at any time
 		DisableStrictNonDeterminismCheck bool
 	}
+
+	// PollerRun is a helper for simpler tracking of the go routine run
+	PollerRun interface {
```

Comment:
nit: move this into a separate file/package instead of one huge internal/worker. This file/package already have too many responsibilities

Reply:
moved